cmd/dp-search-api/main.go

```diff
@@ -92,3 +110,30 @@ func main() {
 
 	os.Exit(1)
 }
+
+func registerCheckers(ctx context.Context,
+	cfg *config.Config,
+	elasticHTTPClient dphttp.Clienter,
+	esSigner *esauth.Signer) *healthcheck.HealthCheck {
+
+	hasErrors := false
+
+	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
+	if err != nil {
+		log.Event(ctx, "error creating version info", log.FATAL, log.Error(err))
+		hasErrors = true
+	}
+
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+
+	elasticClient := elastic.NewClientWithHTTPClientAndAwsSigner(cfg.ElasticSearchAPIURL, esSigner, cfg.SignElasticsearchRequests, elasticHTTPClient)
+	if err = hc.AddCheck("Elasticsearch", elasticClient.Checker); err != nil {
+		log.Event(ctx, "error creating elasticsearch health check", log.ERROR, log.Error(err))
+		hasErrors = true
+	}
+
+	if hasErrors {
+		os.Exit(1)
```

Comment:
Can you add another log event here before exiting please? Something like:

`"failed to successfully register API checks - exiting"` 

Or something along those lines. I know we are logging errors for each of the failed register attempts but I think its probably good to be explicitly clear that the app exit was expected due to the errors rather than it suddenly stopping. Anyone investigating this would then know it was 100% due to the checks failing and not some other issue.